include/xsparse/level_capabilities/co_iteration.hpp

```diff
@@ -0,0 +1,197 @@
+#ifndef XSPARSE_CO_ITERATION_HPP
+#define XSPARSE_CO_ITERATION_HPP
+
+#include <vector>
+#include <tuple>
+#include <algorithm>
+#include <climits>
+
+namespace xsparse::level_capabilities
+{
+    template <class F, class IK, class PK, class Levels, class Is>
+    class Coiterate;
+
+    template <class F, class IK, class PK, class... Levels, class... Is>
+    class Coiterate<F, IK, PK, std::tuple<Levels...>, std::tuple<Is...>>
+    {
+    private:
+        std::tuple<Is...> const m_i;
+        PK const m_pkm1;
+        std::tuple<Levels&...> const m_levelsTuple;
```

Comment:
PKM1 and I should be inside the coiteration helper.

Ideally I'd like coiterate to take levels and helper to take PKM1 and I.

Reply:
Oh yeah, sorry I missed that, I'll make the change.

Reply:
@hameerabbasi but if we move PKM1 and I inside the helper class, then we would also need to pass an instance of the outer `Coiterate` since we need the tuple of levels to create the `iter_helpers` tuple. Would that be okay?

Reply:
Yes, that is desired behaviour. You will have a iter_helper method on Coiterator which should take I and PKM1 and produce a coiteration_helper.